src/mscorlib/src/System/IO/FileStream.cs

```diff
@@ -1358,6 +1358,24 @@ private void FlushOSBuffer()
             }
         }
 
+
+        // This method allows the user to specify an external buffer, which can then be pooled or shared
+        // among mulitple instances of FileStream. Note that this buffer can _only_ be used by a single
+        // instance of FileStream at a time. 
+        // This should be called before any action is done on the FileStream, otherwise a buffer will 
+        // be allocation, and we'll miss this optimization
+        [System.Security.SecuritySafeCritical]  // auto-generated
+        public override void SetBuffer(byte[] externalBuffer)
```

Comment:
Hm, slip of the keyboard while writing the comment. I think that the name SetBuffer is better, with the externalBuffer name indicating what is going on.